okhttp/src/main/java/com/squareup/okhttp/internal/DiskLruCache.java

```diff
@@ -291,13 +291,23 @@ private void readJournal() throws IOException {
       if (!source.exhausted()) {
         rebuildJournal();
       } else {
-        journalWriter = Okio.buffer(fileSystem.appendingSink(journalFile));
+        journalWriter = newJournalWriter();
       }
     } finally {
       Util.closeQuietly(source);
     }
   }
 
+  private BufferedSink newJournalWriter() throws FileNotFoundException {
+    Sink fileSink = fileSystem.appendingSink(journalFile);
+    Sink faultHidingSink = new FaultHidingSink(fileSink) {
+      @Override protected void onException(IOException e) {
+        hasJournalErrors = true;
```

Comment:
(We only write to the journal inside a synchronized block, but I'll make that explicit)